scylla-cql/src/errors.rs

```diff
@@ -40,6 +40,12 @@ pub enum QueryError {
 
     #[error("Unable to allocate stream id")]
     UnableToAllocStreamId,
+
+    #[error("A probable bug in driver: {0}")]
+    DriverLogicError(&'static str),
+
+    #[error("User action caused a malfunction: {0}")]
+    UserLogicError(&'static str),
```

Comment:
We can later decide whether we want DriverLogicError and UserLogicError to be there and push it in a separate series, but I don't think it belongs in this one. DriverLogicError has zero callers, and UserLogicError has one, which I think is not needed. Let's minimize this series by dropping the introduction of these errors here, and create a separate issue where we can have a design discussion